8.0 add module partner_external_maps (module google_map in better !)

alexis-via opened the pull request:
See the module description to get more information about this module.

alexis-via committed:
Add module partner_address_on_map

alexis-via committed:
Add ACL

Add post-install script to set a default value in the users's preferences
Take street2 into account

alexis-via committed:
Add standard OSM map url

alexis-via committed:
Add Here Maps

Convert to https when possible

alexis-via committed:
Uses URL with latitude and longitude when base_geolocalize is installed and the partner has the appropriate info

alexis-via committed:
Add ROUTE urls to show a map with start/destination (Gmaps, OSM and Here)

alexis-via committed:
Rename module to partner_external_maps

Rename object map.url to map.website
User can now choose a different map.website for routing
Code and view cleanup
Add translation files

youring commented:
:thumbsup: Suggestion: would be better to fetch the address_format set in Sales--Configuration--Address Book--Localization--Countries when doing _address_as_string method. Since country, state, city, address sequences may vary on diffrent countries.

alexis-via committed:
Add MapQuest

alexis-via committed:
Add Yahoo! Maps

Enhance bing maps url for lat/lon (now has a pointer)
Update module description

alexis-via committed:
FIX domain declaration

eLBati commented on partner_external_maps/README.rst:
```diff
@@ -0,0 +1,54 @@
+In the old days of Odoo/OpenERP, back in version 6.1, there was an official *google_map* module ; this module added a *Map* button on the partner form view and, when the user clicked on that button, it would open a new tab on its web browser and go to Google Map with a search on the address of the partner.
+
+This module aims at restoring this feature with several improvements:
+* each user can select the map website he wants to use in its preferences
```
Please add newline around '\* '

alexis-via committed:
Add newline around '*' in README.rst, as suggested by Lorenzo

pedrobaeza commented on partner_external_maps/partner_external_maps.py:
```diff
@@ -0,0 +1,197 @@
+
+    @api.model
+    def _default_map_settings(self):
+        '''Method called from post-install script
```
For docstring, use triple double quota instead of single one

pedrobaeza replied:
Although you need the post-init script for the existing users, it still would be useful to provide a default value for new created users since the installation of the module.

pedrobaeza commented on partner_external_maps/partner_external_maps.py:
```diff
@@ -0,0 +1,197 @@
+    @api.model
+    def _address_as_string(self):
+        addr = u''
+        if self.street:
```
Isn't better to add all existing fields to a list and build the string with the `' '.join()` expression?

pedrobaeza commented on partner_external_maps/partner_view.xml:
```diff
@@ -0,0 +1,31 @@
+    <field name="inherit_id" ref="base.view_partner_form"/>
+    <field name="arch" type="xml">
+        <xpath expr="//div[@name='buttons']" position="inside">
+            <button class="oe_inline oe_stat_button"
```
Isn't better to provide these buttons next to the street information or at least inside the same tab instead of being mixed with the bunch of buttons that is already present on the partner in the smart-buttons area?

alexis-via committed:
Add default values when you create a new user

Use ' '.join() to generate the address string
For docstring, use triple double quotes instead of single quotes

alexis-via commented:
@pedrobaeza I took into account your remarks
About the buttons, I think it's a good choice to group them all at the same place ; that's how the form view of the partner form has been designed by Odoo S.A. Of course, we can all have different opinions about this...

pedrobaeza commented:
Thanks :+1:

cdubuit commented:
Merci :+1:

pedrobaeza commented:
Trying on runbot, I'm getting `Error: Unknown field street in domain [["street","=",false],["city","=",false]]` accessing the form view. This should happend when the field is not included in the view. Do you know the case where street field is not included in the view. Is there any module that replaces it?

About @youring suggestion, what do you think?

alexis-via committed:
Remove street = False from domain, to be compatible with partner_street_number

alexis-via commented:
@pedrobaeza it's the module "partner_street_number" that removes the street field from the view. It's not a problem to only keep the city = False in the domain, so the module is now compatible with "partner_street_number"

alexis-via commented:
@pedrobaeza after my last commit, we still have an error msg in runbot: "Error: NameError: name 'street' is not defined". But the issue is not in my module.

I think it is a very bad practise to remove fields from view as they did it in partner_street_number, because it breaks all modules that use the street field in domain or in view inheritance. So I make a PR to fix it:
https://github.com/OCA/partner-contact/pull/135

pedrobaeza commented:
#135 is merged, so you can rebase, Alexis.

alexis-via commented:
I tested runbot after my rebase, and it works fine: no more error msg when you access the form view of a partner !

I think we can merge now.

pedrobaeza commented:
Indeed it's working! Thanks

:+1:

pedrobaeza commented:
Following the singular conventions, I'm going to rename the module to _partner_external_map_

pedrobaeza commented:
Well, thinking twice, we can stay as is as the meaning is not kept. On v9 we can find another name.
